alacritty/src/main.rs

```diff
@@ -180,9 +181,20 @@ fn run(
     // The monitor watches the config file for changes and reloads it. Pending
     // config changes are processed in the main loop.
     if config.ui_config.live_config_reload {
-        monitor::watch(config.ui_config.config_paths.clone(), event_proxy);
+        monitor::watch(config.ui_config.config_paths.clone(), event_proxy.clone());
     }
 
+    // Reload configuration when receiving a SIGUSR1 signal.
+    let config_path = config.ui_config.config_paths[0].clone();
+    std::thread::spawn(move || {
+        let mut signals = Signals::new(&[sigconsts::SIGUSR1]).unwrap();
+        for sig in signals.forever() {
+            if sig == sigconsts::SIGUSR1 {
+                event_proxy.send_event(Event::ConfigReload(config_path.clone()));
+            }
+        }
+    });
```

Comment:
I think it would be nice to make this part of the monitor. Just so we have config reload watching all in one place.

I'd also always prefer spawning a named thread, though I personally think that a separate thread just for this is a bit unfortunate. Especially because we can expect that 99.9% of users will never use, nor even know about this feature, since live config reload exists.

Reply:
> I think it would be nice to make this part of the monitor. Just so we have config reload watching all in one place.

Sure.

> I'd also always prefer spawning a named thread, though I personally think that a separate thread just for this is a bit unfortunate. Especially because we can expect that 99.9% of users will never use, nor even know about this feature, since live config reload exists.

I agree, but I don't really know what other (safe) way there is. I couldn't find a way to plug signal events into the glutin event loop directly. We could make signal_hook set a flag, but then when do we read the flag?

Reply:
Receiving it through the event loop would be ideal, but I'd assume that's probably going to be problematic since it's not really known for its extensibility. At least not without spawning worker threads anyway.

Maybe things could at least be combined with the monitor thread in some way? We already got that one around anyway.

Reply:
Yeah, I'm not sure how. The config watcher thread listens on a channel, and I don't see a straightforward way to have a signal handler wake that channel up.